### ISSUE TITLE
/var/www/flarum not deleted during app removal

### ISSUE THREAD

**Issue by gamersalpha** [YunoHost-Apps/flarum_ynh#53]
Hello i try t install u apps, but i have this error : 
```

`+ true

ERROR 1396 (HY000) at line 1: Operation DROP USER failed for 'flarum'@'localhost'

+ mysql -u root --password=password -B ''

+ ynh_mysql_connect_as root password ''

++ sudo cat /etc/yunohost/mysql

+ ynh_mysql_execute_as_root 'DROP USER '\''flarum'\''@'\''localhost'\'';'

+ ynh_mysql_drop_user flarum

+ true

ERROR 1008 (HY000) at line 1: Can't drop database 'flarum'; database doesn't exist

+ mysql -u root --password=password -B ''

+ ynh_mysql_connect_as root password ''

++ sudo cat /etc/yunohost/mysql

+ ynh_mysql_execute_as_root 'DROP DATABASE flarum;'

+ ynh_mysql_drop_db flarum

+ dbuser=flarum

+ dbname=flarum

+ sudo service php5-fpm reload

+ sudo service nginx reload

+ '[' -e /etc/php5/fpm/pool.d/flarum.conf ']'

+ REMOVE_FPM_CONF

+ '[' -e /etc/nginx/conf.d/domaine.fr.d/flarum.conf ']'

+ REMOVE_NGINX_CONF

+ return 1

+ echo 'No detected variable.'

No detected variable.

+ '[' 0 -eq 1 ']'

+ grep -q '\$'

+ echo /opt/flarum_composer

+ no_var=0

+ chaine=/opt/flarum_composer

+ SECURE_REMOVE /opt/flarum_composer

+ return 1

+ echo 'No detected variable.'

No detected variable.

+ '[' 0 -eq 1 ']'

+ echo /var/www/flarum

+ grep -q '\$'

+ no_var=0

+ chaine=/var/www/flarum

+ SECURE_REMOVE /var/www/flarum

+ return 1

No detected variable.

+ echo 'No detected variable.'

+ '[' 0 -eq 1 ']'

+ grep -q '\$'

+ echo /tmp/composerinstall

+ no_var=0

+ chaine=/tmp/composerinstall

+ SECURE_REMOVE /tmp/composerinstall

+ domain=domaine.fr

++ sudo yunohost app setting flarum domain --output-as plain --quiet

++ ynh_app_setting_get flarum domain

++ . /usr/share/yunohost/helpers.d/utils

++ '[' -r /usr/share/yunohost/helpers.d/utils ']'

++ for helper in '$(run-parts --list /usr/share/yunohost/helpers.d 2>/dev/null)'

++ . /usr/share/yunohost/helpers.d/user

++ '[' -r /usr/share/yunohost/helpers.d/user ']'

++ for helper in '$(run-parts --list /usr/share/yunohost/helpers.d 2>/dev/null)'

++ . /usr/share/yunohost/helpers.d/system

++ '[' -r /usr/share/yunohost/helpers.d/system ']'

++ for helper in '$(run-parts --list /usr/share/yunohost/helpers.d 2>/dev/null)'

++ . /usr/share/yunohost/helpers.d/string

++ '[' -r /usr/share/yunohost/helpers.d/string ']'

++ for helper in '$(run-parts --list /usr/share/yunohost/helpers.d 2>/dev/null)'

++ . /usr/share/yunohost/helpers.d/setting

++ '[' -r /usr/share/yunohost/helpers.d/setting ']'

++ for helper in '$(run-parts --list /usr/share/yunohost/helpers.d 2>/dev/null)'

++ . /usr/share/yunohost/helpers.d/print

++ '[' -r /usr/share/yunohost/helpers.d/print ']'

++ for helper in '$(run-parts --list /usr/share/yunohost/helpers.d 2>/dev/null)'

++ . /usr/share/yunohost/helpers.d/package

++ '[' -r /usr/share/yunohost/helpers.d/package ']'

++ for helper in '$(run-parts --list /usr/share/yunohost/helpers.d 2>/dev/null)'

++ . /usr/share/yunohost/helpers.d/network

++ '[' -r /usr/share/yunohost/helpers.d/network ']'

++ for helper in '$(run-parts --list /usr/share/yunohost/helpers.d 2>/dev/null)'

+++ MYSQL_ROOT_PWD_FILE=/etc/yunohost/mysql

++ . /usr/share/yunohost/helpers.d/mysql

++ '[' -r /usr/share/yunohost/helpers.d/mysql ']'

++ for helper in '$(run-parts --list /usr/share/yunohost/helpers.d 2>/dev/null)'

++ . /usr/share/yunohost/helpers.d/ip

++ '[' -r /usr/share/yunohost/helpers.d/ip ']'

++ for helper in '$(run-parts --list /usr/share/yunohost/helpers.d 2>/dev/null)'

+++ CAN_BIND=1

++ . /usr/share/yunohost/helpers.d/filesystem

++ '[' -r /usr/share/yunohost/helpers.d/filesystem ']'

++ for helper in '$(run-parts --list /usr/share/yunohost/helpers.d 2>/dev/null)'

++ . /usr/share/yunohost/helpers.d/backend

++ '[' -r /usr/share/yunohost/helpers.d/backend ']'

++ for helper in '$(run-parts --list /usr/share/yunohost/helpers.d 2>/dev/null)'

+++ run-parts --list /usr/share/yunohost/helpers.d

+ source /usr/share/yunohost/helpers

+ app=flarum

Exécution du script « /var/cache/yunohost/from_file/flarum_ynh-master/scripts/remove »...

+ exit 1

+ echo ''

+ ynh_die

+ sudo sed -i '\@"domaine.fr/flarum/":@d' /etc/ssowat/conf.json

+ type -t CLEAN_SETUP

+ echo -e '!!\n flarum install'\''s script has encountered an error. Installation was cancelled.\n!!'

[91m [1m

+ echo -e '\e[91m \e[1m'

+ set +eu

+ trap '' EXIT

+ '[' 1 -eq 0 ']'

+ exit_code=1

!!

+ EXIT_PROPERLY

flarum install's script has encountered an error. Installation was cancelled.

+ false

!!

+ echo 'This path already contains a folder'

+ '[' -e /var/www/flarum ']'

This path already contains a folder

+ final_path=/var/www/flarum

+ CHECK_FINALPATH

+ sudo yunohost app setting flarum path --value=/flarum --quiet

+ ynh_app_setting_set flarum path /flarum

+ sudo yunohost app setting flarum is_public --value=1 --quiet

+ ynh_app_setting_set flarum is_public 1

+ sudo yunohost app setting flarum admin --value=user01 --quiet

+ ynh_app_setting_set flarum admin user01

'yunohost app checkurl' is deprecated and will be removed in the future

+ sudo yunohost app checkurl domaine.fr/flarum -a flarum

+ CHECK_DOMAINPATH

+ '[' m == / ']'

+ '[' / '!=' / ']'

+ CHECK_PATH

+ grep -q '"username": "user01"'

+ sudo yunohost user list --output-as json

+ ynh_user_exists user01

+ CHECK_USER user01

+ test -n flarum

+ CHECK_VAR flarum 'app name not set'

++ . /usr/share/yunohost/helpers.d/utils

++ '[' -r /usr/share/yunohost/helpers.d/utils ']'

++ for helper in '$(run-parts --list /usr/share/yunohost/helpers.d 2>/dev/null)'

++ . /usr/share/yunohost/helpers.d/user

++ '[' -r /usr/share/yunohost/helpers.d/user ']'

++ for helper in '$(run-parts --list /usr/share/yunohost/helpers.d 2>/dev/null)'

++ . /usr/share/yunohost/helpers.d/system

++ '[' -r /usr/share/yunohost/helpers.d/system ']'

++ for helper in '$(run-parts --list /usr/share/yunohost/helpers.d 2>/dev/null)'

++ . /usr/share/yunohost/helpers.d/string

++ '[' -r /usr/share/yunohost/helpers.d/string ']'

++ for helper in '$(run-parts --list /usr/share/yunohost/helpers.d 2>/dev/null)'

++ . /usr/share/yunohost/helpers.d/setting

++ '[' -r /usr/share/yunohost/helpers.d/setting ']'

++ for helper in '$(run-parts --list /usr/share/yunohost/helpers.d 2>/dev/null)'

++ . /usr/share/yunohost/helpers.d/print

++ '[' -r /usr/share/yunohost/helpers.d/print ']'

++ for helper in '$(run-parts --list /usr/share/yunohost/helpers.d 2>/dev/null)'

++ . /usr/share/yunohost/helpers.d/package

++ '[' -r /usr/share/yunohost/helpers.d/package ']'

++ for helper in '$(run-parts --list /usr/share/yunohost/helpers.d 2>/dev/null)'

++ . /usr/share/yunohost/helpers.d/network

++ '[' -r /usr/share/yunohost/helpers.d/network ']'

++ for helper in '$(run-parts --list /usr/share/yunohost/helpers.d 2>/dev/null)'

+++ MYSQL_ROOT_PWD_FILE=/etc/yunohost/mysql

++ . /usr/share/yunohost/helpers.d/mysql

++ '[' -r /usr/share/yunohost/helpers.d/mysql ']'

++ for helper in '$(run-parts --list /usr/share/yunohost/helpers.d 2>/dev/null)'

++ . /usr/share/yunohost/helpers.d/ip

++ '[' -r /usr/share/yunohost/helpers.d/ip ']'

++ for helper in '$(run-parts --list /usr/share/yunohost/helpers.d 2>/dev/null)'

+++ CAN_BIND=1

++ . /usr/share/yunohost/helpers.d/filesystem

++ '[' -r /usr/share/yunohost/helpers.d/filesystem ']'

++ for helper in '$(run-parts --list /usr/share/yunohost/helpers.d 2>/dev/null)'

++ . /usr/share/yunohost/helpers.d/backend

++ '[' -r /usr/share/yunohost/helpers.d/backend ']'

++ for helper in '$(run-parts --list /usr/share/yunohost/helpers.d 2>/dev/null)'

+++ run-parts --list /usr/share/yunohost/helpers.d

+ source /usr/share/yunohost/helpers

+ final_path=/var/www/flarum

+ www_path=/var/www

+ is_public=1

+ title='Forum domaine'

+ admin=user01

+ path=/flarum

+ domain=domaine.fr

+ app=flarum

+ trap EXIT_PROPERLY EXIT

+ set -eu

+ TRAP_ON

+ source .functions

Exécution du script « /var/cache/yunohost/from_file/flarum_ynh-master/scripts/install »...

Vérification des paquets requis pour flarum...

Terminé

Extraction...

Téléchargement...

Connecté

Non autorisé`

```
what do u think please

**Comment by frju365** [comment]
it seems that the /var/www/flarum already exists, and, so, it can't create that again. Could you remove /var/www/flarum if it exists. But this app is inprogress for now and doesn't work yet. There is a bug in LDAP extension.

**Comment by tituspijean** [comment]
Confirmed on my side with two instances of the app.
There seems to be an issue with `SECURE_REMOVE`, or at least the deletion of the `/var/www/{$app}` folder is not performed during app removal.

**Comment by tituspijean** [comment]
Fixed by #55.
